ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -63,44 +58,92 @@ impl Distribution<f64> for RoundedBoxMuller {
         self.inner.sample(rng).round()
     }
 }
-
 impl From<BoxMuller> for RoundedBoxMuller {
     fn from(value: BoxMuller) -> Self {
         Self { inner: value }
     }
 }
-
+/// Double Geometric Distribution
+///
+/// Generates a sample from a geometric distribution with the given success probability.
+fn generate_geometric<R: Rng + ?Sized>(probability: f64, rng: &mut R) -> isize {
```

Comment:
this is a bit different than a geometric distribution implemented [here](https://docs.rs/rand_distr/latest/src/rand_distr/geometric.rs.html#32-37). If this is intentional, I would still recommend to follow their pattern, i.e. define a struct and implement `Distribution<u64>` for it

Reply:
defined a struct and implemented Disitribution<u32> for it